### PR TITLE
Config rewrite updates

### PR DESCRIPTION
**Pull request by sirainen**
DOV-6027

### CODE REVIEW

File: source/configuration_manual/authentication/password_database_extra_fields.rst [hunk 1]
```diff
@@ -33,7 +33,7 @@ specially:
 
   .. dovecotadded:: 2.2.22
 
-* **k5principals**: if using ``auth_mechanisms = gssapi`` , may contain
+* **k5principals**: if using :dovecot_core:ref:`auth_mechanisms` = gssapi`, may contain
```
Comment (cmouse) [inline source/configuration_manual/authentication/password_database_extra_fields.rst:36]: extra `?

**Review by cmouse — CHANGES_REQUESTED** [review]

File: source/settings/core.rst [hunk 2]
```diff
@@ -2866,12 +2866,9 @@ See :ref:`settings` for list of all setting groups.
...
 .. dovecot_core:setting:: protocols
    :default: imap pop3 lmtp
-   :values: @string
-
-   The list of protocols this node will support.
+   :values: @boollist
 
-   It takes a space-separated list of protocols (which are configured
-   separately) as its value.
+   The list of protocols to enable.
```
Comment (cmouse) [inline source/settings/core.rst:2871]: so how do you enable **just** one protocol?

**Review by cmouse — CHANGES_REQUESTED** [review]

File: source/configuration_manual/event_export.rst [hunk 3]
```diff
@@ -161,6 +161,8 @@ Example tab-text
 
 .. _event_exporter_transports:
 
+.. _event_exporter_transports:
```
[the remark this answers is not in the document]
Comment (sirainen) [reply source/configuration_manual/event_export.rst:164]: Annoyingly it keeps coming back in rebases.

File: source/settings/core.rst (hunk 2 above)
Comment (sirainen) [reply source/settings/core.rst:2871]: Same as before, `protocols = imap` - list can be just one. So not sure what could be clarified here?

Comment (cmouse) [reply source/settings/core.rst:2871]: it could show an example here for both. because now all around documentation we only have the incremental one.

Comment (sirainen) [reply source/settings/core.rst]: How about now?

**Review by cmouse — CHANGES_REQUESTED** [review]

**Pull request merged**